### PR TITLE
Add referral componet to display referral on patient chart

### PR DESCRIPTION
**Pull request by makombe**
## Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Screenshots
Screenshot from 2024-03-26 09-12-47

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

**Comment by makombe** [comment]
One more thing remaining is how to get the nupi from patient chart. Currently I have hard coded the nupi number I was using for testing. This will change. Added a comment a long side it as a TODO cc @ojwanganto , @jecihjoy

File: packages/esm-shr-app/src/community-referrals/patient-chart/referral-chart-view.component.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import { useCommunityReferral } from '../community-refferals.resource';
+
+export interface ReferralReasonsDialogPopupProps {
+  patientUuid: string;
+}
+
+const ReferralReasonsView: React.FC<ReferralReasonsDialogPopupProps> = ({ patientUuid }) => {
+  const { t } = useTranslation();
+  const layout = useLayoutType();
+  const { referral, isLoading, isValidating } = useCommunityReferral('MOH8DW535506J'); // TODO dynamically fetch nupi from patient chart
```
Comment (makombe) [inline packages/esm-shr-app/src/community-referrals/patient-chart/referral-chart-view.component.tsx:16]: This is the part we will be changing.

File: packages/esm-shr-app/src/types/index.ts [hunk 2]
```diff
@@ -188,11 +188,13 @@ export interface CommunityReferral {
   birthdate: string;
   gender: string;
   referralReasons: ReferralReasonsProps;
+  status?: string;
 }
 
 export interface ReferralReasonsProps {
```
Comment (jecihjoy) [inline packages/esm-shr-app/src/types/index.ts:194]: @makombe it would be useful to show where patient was referred from

Comment (makombe) [reply packages/esm-shr-app/src/types/index.ts:194]: @jecihjoy This has been adressed.Thanks

File: packages/esm-shr-app/src/dashboard.meta.ts [hunk 3]
```diff
@@ -1,6 +1,15 @@
...
+//   path: 'SHR panel',
+// };
+
+export const communityReferralDashboardMeta = {
+  slot: 'patient-chart-referral-slot',
   columns: 1,
-  title: 'SHR panel',
-  path: 'SHR panel',
+  title: 'Referrals',
+  path: 'community-referral',
```
Comment (ojwanganto) [inline packages/esm-shr-app/src/dashboard.meta.ts:12]: I think we can change this to referrals. We intend to use the same component for both referrals from community and facility

**Review by ojwanganto — APPROVED** [review]

**Pull request merged**